### PR TITLE
doc: promote SPIFFE Cert Validator to alpha, set security_posture.

### PR DESCRIPTION
**Pull request by mathetake**
resolves https://github.com/envoyproxy/envoy/issues/17511.

cc @lizan @azdagron @evan2645

Signed-off-by: Takeshi Yoneda <takeshi@tetrate.io>

### CODE REVIEW

**Review by phlax — APPROVED** [review]
lgtm, thanks @mathetake 

cc @htuch for final stamp

File: source/extensions/extensions_metadata.yaml [hunk 1]
```diff
@@ -592,8 +592,8 @@ envoy.stat_sinks.wasm:
 envoy.tls.cert_validator.spiffe:
   categories:
   - envoy.tls.cert_validator
-  security_posture: unknown
-  status: wip
+  security_posture: robust_to_untrusted_downstream_and_upstream
```
Comment (htuch) [inline source/extensions/extensions_metadata.yaml:595]: Do we have a fuzzer for this? I'm not confident at a glance that the error checking from BoringSSL is robust and I think for something that is dealing with security sensitive cert validation a fuzzer would be best.

Comment (mathetake) [reply source/extensions/extensions_metadata.yaml:595]: we don't have fuzzer on this, so I will put `requires_trusted_downstream_and_upstream` for now, and after having fuzzer, I will raise the PR again to change to `robust_to_untrusted_downstream_and_upstream`. sg?

Comment (htuch) [reply source/extensions/extensions_metadata.yaml:595]: Sure

**Review by htuch — APPROVED** [review]
LGTM, thanks!

**Comment by azdagron** [comment]
Thanks for this, @mathetake !

**Pull request merged**